src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/LæremidlerBeregnYtelseSteg.kt

```diff
@@ -62,32 +69,63 @@ class LæremidlerBeregnYtelseSteg(
         lagreAndeler(saksbehandling, beregningsresultat)
     }
 
-    /*
-    private fun beregnOgLagreOpphør(saksbehandling: Saksbehandling, vedtak: OpphørLæremidlerDto) {
+    private fun beregnOgLagreOpphør(saksbehandling: Saksbehandling, vedtak: OpphørLæremidlerRequest) {
         brukerfeilHvis(saksbehandling.forrigeBehandlingId == null) {
             "Opphør er et ugyldig vedtaksresultat fordi behandlingen er en førstegangsbehandling"
         }
+        val forrigeBehandling = vedtakRepository.findByIdOrNull(saksbehandling.forrigeBehandlingId)
+            ?: error("Opphør må ha en forrige behandling")
+        if (saksbehandling.revurderFra == null) {
+            error("revurderFra-dato er påkrevd for opphør")
+        }
 
-        opphørValideringService.validerPerioder(saksbehandling)
+        val kuttedePerioder: List<BeregningsresultatForMåned> = kuttePerioderVedOpphør(forrigeBehandling, saksbehandling.revurderFra)
 
-        val beregningsresultat = tilsynBarnBeregningService.beregn(saksbehandling, TypeVedtak.OPPHØR)
-        opphørValideringService.validerIngenUtbetalingEtterRevurderFraDato(beregningsresultat, saksbehandling.revurderFra)
         vedtakRepository.insert(
             GeneriskVedtak(
                 behandlingId = saksbehandling.id,
                 type = TypeVedtak.OPPHØR,
                 data = OpphørLæremidler(
-                    beregningsresultat = BeregningsresultatLæremidler(beregningsresultat.perioder),
+                    beregningsresultat = BeregningsresultatLæremidler(kuttedePerioder),
                     årsaker = vedtak.årsakerOpphør,
                     begrunnelse = vedtak.begrunnelse,
                 ),
-
-                ),
+            ),
         )
 
-        lagreAndeler(saksbehandling, beregningsresultat)
+        val beregningsresultatLæremidler = BeregningsresultatLæremidler(kuttedePerioder)
+
+        lagreAndeler(saksbehandling, beregningsresultatLæremidler)
+    }
+
+    private fun kuttePerioderVedOpphør(forrigeBehandling: Vedtak, revurderFra: LocalDate): List<BeregningsresultatForMåned> {
```

Comment:
```suggestion
    private fun kuttePerioderVedOpphør(forrigeVedtak: Vedtak, revurderFra: LocalDate): List<BeregningsresultatForMåned> {
```

Forrige vedtak kommer alltid være innvilgelse eller opphør, og inneholder alltid et nåbilde av alle vedtaksperioder og beregningsperioder